float32[] translation
float32[] rotation
